pkg/handlers/deploy.go

```diff
@@ -42,10 +42,22 @@ func MakeDeployHandler(functionNamespace string, factory k8s.FunctionFactory) ht
 			defer r.Body.Close()
 		}
 
+		hasProfiles, err := k8s.ProfilesEnabled(factory.Client)
```

Comment:
Previously, would this handler work if the CRD wasn't present and we had no annotation for profiles in the request?

Reply:
yes, if there are no profile annotations present, then `profileList, err = factory.GetProfiles` on L90 should return an empty list as a noop, there are various spots in the profiles code that will just be empty loops.